test/e2e/redundancy/gateway_failover.go

```diff
@@ -27,9 +37,19 @@ var _ = Describe("[redundancy] Gateway fail-over tests", func() {
 			testGatewayFailOverScenario(f)
 		})
 	})
-
 })
 
+func testBasicGatewayReporting(f *subFramework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+
+	By(fmt.Sprintf("Ensuring that one gateway reports as active %q", clusterAName))
+	activeGateways := f.GetGatewaysWithHAStatus(subv1.HAStatusActive)
+	Expect(activeGateways).To(HaveLen(1))
```

Comment:
I think this should be wrapped with `framework.AwaitUntil` to avoid intermittent failures if the active was not yet running or written it's status. Perhaps `AwaitActiveGateway`.

Reply:
Actually there's already `AwaitGatewayWithStatus` although it takes the gateway name which could be obtained by finding the active gateway node. But I realize we're actually duplicating what `testEnginePodRestartScenario` does so why do we need this test?

Reply:
For independence, I wanted an indication that the basic reporting is working if the dataplane is broken. When we have stats then we could identify if it's the dataplane breaking, or the reporting breaking. The reporting would always affect the dataplane tests though.

Reply:
I think we should do AwaitActiveGateway at least to avoid possible intermittent failures as you said.

Reply:
Oh AwaitGateway does not work, because we don't know the name, but we know that we want one. I will add another function that works with any.